tests/integration/define.py

```diff
@@ -0,0 +1,324 @@
+from conditions import (
+    ProvenanceCond,
+    RegexCond,
+    ReturnCodeCond,
+    StepFileContainsCond,
+    StepFileExistsCond,
+)
+
+from merlin.utils import get_flux_cmd
+
+
+OUTPUT_DIR = "cli_test_studies"
+
+
+def define_tests():
+    """
+    Returns a dictionary of tests, where the key
+    is the test's name, and the value is a tuple
+    of (shell command, condition(s) to satisfy).
+    """
+    celery_regex = r"(srun\s+.*)?celery\s+(-A|--app)\s+merlin\s+worker\s+.*"
+
+    # shortcut string variables
+    workers = "merlin run-workers"
+    run = "merlin run"
+    restart = "merlin restart"
+    purge = "merlin purge"
+    examples = "merlin/examples/workflows"
+    dev_examples = "merlin/examples/dev_workflows"
+    demo = f"{examples}/feature_demo/feature_demo.yaml"
+    demo_pgen = f"{examples}/feature_demo/scripts/pgen.py"
+    simple = f"{examples}/simple_chain/simple_chain.yaml"
+    slurm = f"{examples}/slurm/slurm_test.yaml"
+    slurm_restart = f"{examples}/slurm/slurm_par_restart.yaml"
+    flux = f"{examples}/flux/flux_test.yaml"
+    flux_restart = f"{examples}/flux/flux_par_restart.yaml"
+    lsf = f"{examples}/lsf/lsf_par.yaml"
+    black = "black --check --target-version py36"
+    config_dir = "./CLI_TEST_MERLIN_CONFIG"
+
+    return {
```

Comment:
I think this file could be significantly improved even more by breaking up the dictionary into thematic sub dictionaries and then grouping them together. For instance we could so something like:

```
merlin_checks = {"merlin": ..., "merlin help": ..., "merlin version": ..., "merlin config": ...}
local_workflows = {"local minimum_format:", ..., etc}
worker_command_checks = {"run-workers echo simple_chain:", ..., etc}
variable_override_checks = {...}

all_tests = {}
for d in [merlin_checks, local_workflows, worker_command_checks, variable_override_checks,]:
    _ = all_tests.update(d)

return all_tests
```